app/code/Magento/Customer/Controller/Address/FormPost.php

```diff
@@ -197,8 +206,24 @@ public function execute()
         }
 
         try {
+            $addressId = $this->getRequest()->getParam('id');
+            $customerId = $this->_getSession()->getCustomerId();
+
+            $customer = $this->customerRepository->getById($customerId);
+
             $address = $this->_extractAddress();
-            $this->_addressRepository->save($address);
+
+            $addresses = [];
+            foreach ($customer->getAddresses() as $customerAddress) {
```

Comment:
How about using `array_filter()` in this specific case?

Reply:
Make sense, thanks!